src/scipp/core/math.py

```diff
@@ -175,3 +175,50 @@ def erfc(x: VariableLike) -> VariableLike:
     Computes the complementary error function.
     """
     return _cpp.erfc(x)
+
+
+def midpoints(x: _cpp.Variable, dim: Optional[str] = None) -> _cpp.Variable:
+    """
+    Computes the points in the middle of adjacent elements of x.
+
+    The midpoint of two numbers :math:`a` and :math:`b` is
+    :math:`(a + b) / 2`.
+    This formula encounters under- or overflow for
+    very small or very large inputs.
+    The implementation deals with those cases properly.
```

Comment:
Wasn't there the opposite comment above?

Reply:
What do you mean? Which comment?

Reply:
```
    // This is *not* safe from over-/underflow!
    // But those should be rare as currently, most data uses reasonably
    // normalized double precision values.
    return (a + b) / 2;
```
whereas here says it deals with those cases properly?

Reply:
Ah right.  I can update the comment to mentions floats specifically.
Or I use an implementation that handles floats correctly. Which do you prefer?

Reply:
Didn't you recently point me to a safer implementation? Why was it not used here?

Reply:
It was only added in C++20. That is why in `core/element/math.h`, there is a check for the C++ standard. I took the integer implementation from the GNU standard library. But I didn't want to copy the entire thing.
I also checked out the LLVM implementation and it is essentially the same as in GNU. So I am simply implementing it for floats based on them now. It won't be a straight up copy, so we should be fine in terms of licensing.

Reply:
👍 I don't really mind either solution. If we had such extreme cases many other things would likely break anyway.